cohere/client.py

```diff
@@ -503,21 +504,83 @@ def create_cluster_job(
         threshold: Optional[float] = None,
         min_cluster_size: Optional[int] = None,
     ) -> CreateClusterJobResponse:
+        """Create clustering job.
+
+        Args:
+            embeddings_url (str): File with embeddings to cluster.
+            threshold (Optional[float], optional): TODO. Defaults to None.
+            min_cluster_size (Optional[int], optional): Minimum number of elements in a cluster. Defaults to None.
+
+        Returns:
+            CreateClusterJobResponse: Created clustering job handler
+        """
+
         json_body = {
             "embeddings_url": embeddings_url,
             "threshold": threshold,
             "min_cluster_size": min_cluster_size,
         }
 
         response = self.__request(cohere.CLUSTER_JOBS_URL, json=json_body)
-        return CreateClusterJobResponse(**response)
+        return CreateClusterJobResponse(
+            job_id=response['job_id'],
+            poll_fn=self.poll_cluster_job,
+        )
 
     def get_cluster_job(
         self,
         job_id: str,
     ) -> GetClusterJobResponse:
+        """Get clustering job results.
+
+        Args:
+            job_id (str): Clustering job id.
+
+        Raises:
+            ValueError: "job_id" is empty
+
+        Returns:
+            GetClusterJobResponse: Clustering job results.
+        """
+
         if not job_id.strip():
             raise ValueError('"job_id" is empty')
 
         response = self.__request(os.path.join(cohere.CLUSTER_JOBS_URL, job_id), method='GET')
-        return GetClusterJobResponse(**response)
+        return GetClusterJobResponse(
+            status=response['status'],
+            output_clusters_url=response['output_clusters_url'],
+            output_outliers_url=response['output_outliers_url'],
+        )
+
+    def poll_cluster_job(
+        self,
+        job_id: str,
+        timeout: float = 0,
+        interval: float = 10,
+    ) -> GetClusterJobResponse:
+        """Poll clustering job results.
+
+        Args:
+            job_id (str): Clustering job id.
+            timeout (float, optional): Poll timeout in seconds. Defaults to 0.
```

Comment:
why set 0 by default here? it's not intuitive to me, might be good to justify?

Reply:
agree, updated to be `None` when no timeout, and added description to docstring